x-pack/plugins/security/server/authentication/authenticator.test.ts

```diff
@@ -1872,6 +1872,25 @@ describe('Authenticator', () => {
       expect(mockOptions.session.invalidate).not.toHaveBeenCalled();
     });
 
+    it('if session does not exist and providers is empty, returns whatever authentication provider returns.', async () => {
```

Comment:
__nit:__ we don't really have a provider :)
```suggestion
    it('if session does not exist and providers is empty, redirects to default logout path.', async () => {
```

Also in this test suite we can actually properly similate the case we're testing:
```ts
      const request = httpServerMock.createKibanaRequest();

      mockOptions = getMockOptions({
        providers: { basic: { basic1: { order: 0, enabled: false } } },
      });
      authenticator = new Authenticator(mockOptions);

      await expect(authenticator.logout(request)).resolves.toEqual(
        DeauthenticationResult.redirectTo(
          '/mock-server-basepath/security/logged_out?msg=LOGGED_OUT'
        )
      );

      expect(mockBasicAuthenticationProvider.logout).not.toHaveBeenCalled();
      expect(mockOptions.session.invalidate).not.toHaveBeenCalled();
```